components/patina_smbios/src/manager.rs

```diff
@@ -0,0 +1,2398 @@
+}
+
+unsafe impl Sync for GlobalSmbiosManager {}
+
+impl GlobalSmbiosManager {
+    const fn new() -> Self {
+        Self { manager: UnsafeCell::new(None) }
+    }
+
+    /// Initialize the global manager with TplMutex protection
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure this is called only once during system initialization
+    unsafe fn initialize(
+        &self,
+        tpl_mutex: TplMutex<'static, SmbiosManager, StandardBootServices>,
+    ) -> Result<(), SmbiosError> {
+        let ptr = self.manager.get();
+        if unsafe { (*ptr).is_some() } {
+            return Err(SmbiosError::InvalidParameter); // Already initialized
+        }
+        unsafe { *ptr = Some(tpl_mutex) };
+        Ok(())
+    }
+
+    /// Get a reference to the TplMutex (returns None if not initialized)
+    ///
+    /// # Safety
+    ///
+    /// Returns a raw reference to the TplMutex. Caller must call .lock()
+    /// to get TPL-protected access to the manager.
+    unsafe fn get(&self) -> Option<&'static TplMutex<'static, SmbiosManager, StandardBootServices>> {
+        unsafe { (*self.manager.get()).as_ref() }
+    }
+
+    /// Clear the manager (for cleanup on error)
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure this is only called during error cleanup
+    unsafe fn clear(&self) {
+        unsafe { *self.manager.get() = None };
+    }
+}
+
+static SMBIOS_MANAGER: GlobalSmbiosManager = GlobalSmbiosManager::new();
+
+/// Storage for the protocol interface pointer (for lifetime management)
+struct GlobalProtocolInterface {
+    interface: UnsafeCell<*mut c_void>,
+}
+
+unsafe impl Sync for GlobalProtocolInterface {}
+
+impl GlobalProtocolInterface {
+    const fn new() -> Self {
+        Self { interface: UnsafeCell::new(core::ptr::null_mut()) }
+    }
+
+    unsafe fn set(&self, ptr: *mut c_void) {
+        unsafe { *self.interface.get() = ptr };
+    }
+
+    #[allow(dead_code)]
+    unsafe fn get(&self) -> *mut c_void {
+        unsafe { *self.interface.get() }
+    }
+
+    unsafe fn clear(&self) {
+        unsafe { *self.interface.get() = core::ptr::null_mut() };
+    }
+}
+
+static SMBIOS_PROTOCOL_INTERFACE: GlobalProtocolInterface = GlobalProtocolInterface::new();
+
+/// Storage for the protocol handle
+struct GlobalProtocolHandle {
+    handle: UnsafeCell<efi::Handle>,
+}
+
+unsafe impl Sync for GlobalProtocolHandle {}
+
+impl GlobalProtocolHandle {
+    const fn new() -> Self {
+        Self { handle: UnsafeCell::new(core::ptr::null_mut()) }
+    }
+
+    unsafe fn set(&self, h: efi::Handle) {
+        unsafe { *self.handle.get() = h };
+    }
+
+    #[allow(dead_code)]
+    unsafe fn get(&self) -> efi::Handle {
+        unsafe { *self.handle.get() }
+    }
+}
+
+static SMBIOS_PROTOCOL_HANDLE: GlobalProtocolHandle = GlobalProtocolHandle::new();
+
+/// Wrapper for static SMBIOS header buffer that implements Sync
+///
+/// SAFETY: This is safe because UEFI DXE runs in a single-threaded environment,
```

Comment:
Technically, DXE is signle-threaded with interrupts. so you still have to guard against an interrupt coming in during a critical section. That's what the TplMutex locks are for...